### ISSUE TITLE
Feature request: a simple CLI only version to generate reports on embedded platforms

### ISSUE THREAD

**Issue by zeno-endemann-kdab** [SaschaWillems/VulkanCapsViewer#147]
It would be very nice to have a simplified CLI version, ideally without any dependencies other than Vulkan, so that it is very easy to compile it for embedded platforms (such as i.MX8) to just generate a report. Would be enough if that report just gets dumped into a file, and uploaded via the regular app. Or something like that.

**Comment by zeno-endemann-kdab** [comment]
Right. It is just, the more easy/convenient it is to cross compile a version for an embedded board, the more people are willing to go through the hassle to do that to submit reports. Not all cross-SDKs will have Qt available.
I have not looked in detail what information a report contains, but I assume it is significantly more than what `vulkaninfo -j` outputs, right? Or would it be possible to just import that output (generated on the embedded device) into the desktop app somehow? Or this simplified version could potentially be just a shell script that enriches the vulkaninfo output enough to have all the necessary info to generate a report on the desktop... Just throwing out some ideas

**Comment by galkinvv** [comment]
For CLI-only version aarch64 linux build also would be useful.

I've tested vulkan compute on Nvidia jetson with ubuntu 18.04 - it works without problems. But I as most jetson users didn't have GUI access to the device, only ssh. Preobuilt or simple-to-build CLI version can allow getting more tegra-based results into database.

**Comment by gary-sweet** [comment]
+1 for me on this. I'd love to be able to submit reports from our embedded hardware, but building Qt on those will be a serious pain.

A simple CLI version that doesn't depend on Qt would be a real help.  I can see that the networking stuff might be trickier without Qt; perhaps if it just dumped a file that could be manually submitted via a webpage (or desktop app as suggested above) you could avoid any networking in that case.

**Comment by SaschaWillems** [comment]
I'm actively looking into this. The command line app will be as light as possible in regards to external dependencies and will not require Qt. See https://github.com/SaschaWillems/VulkanCapsViewerCmdLine

**Comment by zeno-endemann-kdab** [comment]
Thanks! Unfortunately it has been quite a while since I last worked on a Vulkan-enabled embedded device, so it may take some time before I get to try it again. Oh well. I have compiled and run your new tool on my desktop just to check, seems to work fine.
